solang-parser/src/lexer.rs

```diff
@@ -796,20 +807,22 @@ impl<'input> Lexer<'input> {
                                             }
                                         }
 
-                                        return Some(Err(
+                                        self.errors.push(
                                             LexicalError::InvalidCharacterInHexLiteral(
                                                 Loc::File(self.file_no, i, i + 1),
                                                 ch,
                                             ),
-                                        ));
+                                        );
+                                        return None;
```

Comment:
The purpose of this change is to keep going when you see an error. `return None;` means "end of file", so this doesn't work. Now if I have a file like so:

```solidity
hex"g"

contract foo {}
```
Now the lexer gives up after the `g`. I think you need `continue;` here, so it goes back to the top level loop in the next function.